filter.go

```diff
@@ -20,69 +19,62 @@ var (
 )
 
 type FilterMatch struct {
-	Name  string
-	Label map[string]string
+	Name   string
+	Labels map[string]string
 }
 
 func LogEvent(event *v1.Event, er *EventRouter) []FilterMatch {
 	var matches []FilterMatch
 	eventRouter = er
+	if er.Config == nil {
+		return matches
+	}
 
+OUTER:
 	for _, metric := range er.Config.Metrics {
-		ret := true
 
 		for _, filter := range metric.EventMatcher {
 			value, err := GetValueFromStruct(event, filter.Key)
+			if err != nil {
+				glog.Errorf("Could not get value for key %s: %v", filter.Key, err)
+				continue OUTER
+			}
 
 			if filter.Expr != "" && err == nil {
-				match, _ := regexp.MatchString(
-					filter.Expr,
-					value,
-				)
-
+				match := filter.regex.MatchString(value)
 				glog.V(5).Infof("Expression: %s Value: %s Match: %v\n", filter.Expr, value, match)
-
-				ret = ret && match
-
-				if !ret {
-					break
+				if !match {
+					continue OUTER
 				}
 			}
-
-			if err != nil {
-				glog.Errorf("Could not get filtered value: %v", err)
-				ret = false
-				break
-			}
 		}
 
-		if ret {
-			var l = make(map[string]string)
-
-			for labelKey, labelSpec := range metric.Labels {
-				value := ""
-				var err error
-				var pod *v1.Pod
-
-				if strings.HasPrefix(labelSpec, PodVirtualTypePrefix) && event.InvolvedObject.Kind == "Pod" {
-					pod, err = getPodObjectForEvent(event)
-					if err == nil {
-						value, err = GetValueFromStruct(pod, strings.Replace(labelSpec, PodVirtualTypePrefix, "", 1))
-					}
-				} else {
-					value, err = GetValueFromStruct(event, labelSpec)
-				}
+		var l = make(map[string]string)
+
+		for labelKey, labelSpec := range metric.Labels {
+			value := ""
+			var err error
+			var pod *v1.Pod
 
-				if err != nil {
-					glog.Errorf("Could not get metric label: %v", err)
-					break
+			if strings.HasPrefix(labelSpec, PodVirtualTypePrefix) && event.InvolvedObject.Kind == "Pod" {
+				pod, err = getPodObjectForEvent(event)
+				if err == nil {
+					value, err = GetValueFromStruct(pod, strings.Replace(labelSpec, PodVirtualTypePrefix, "", 1))
 				}
+			} else {
+				value, err = GetValueFromStruct(event, labelSpec)
+			}
 
-				l[labelKey] = value
+			if err != nil {
+				glog.Errorf("Could not get metric label: %v", err)
+				//TODO: Shouldn't this be a continue?
+				break
```

Comment:
@databus23 Maybe we should skip the whole metric if there is a problem with getting the labels?

Reply:
My point is a `break` in this place will leave the outer for loop, basically exiting from processing any more metrics that came later in the array. A `continue` would skip the metric with missing labels but continue to process later metrics. Correct?

Reply:
As far as i see `break` will only exit the inner for loop and will continue on the next metric. Problem is that labels are not well defined and depend on involved objects.

Reply:
But the thing is we are already in the outer loop here.

Reply:
a no, ok I see know, there is another for loop. ok then a break will only skip the label. Which is probably not what we want a `continue OUTER` would be better to skip the metric altogether. I will write a test case for that